xkeysnail/xlib_info.py

```diff
@@ -2,60 +2,53 @@
 
 """Modules with methods related to xlib."""
 
-import os
-import sys
-import Xlib.display
+import Xlib
+from Xlib.display import Display
 
 from xkeysnail.log import log_msg
 
 
-def get_display(display=None, time=10):
-    """Function to wait and get Display."""
-    log_count = False
-    while not time == 0:
-        try:
-            display = Xlib.display.Display()
-            return display
-        # Xlib.error.DisplayConnectionError
-        # This solution is under test, I don't know when and why the
-        # error occurs, my theory is that during boot, it is not possible
-        # to connect to the display because it is not available, waiting
-        # for the display can be a solution.
-        except Xlib.error.DisplayConnectionError as Display_error:
-            if not log_count:
-                log_count = True
-                log_msg(Display_error)
-            os.system("xhost +SI:localuser:root")
-            time =- 1
-    return None
-
-
-def get_focused_window(display=get_display()):
-    if not display:
-        sys.exit(0)
-    return display.get_input_focus().focus
-
-
-def get_window_info(window=get_focused_window()):
-    """Get all info from focused window's (recursively checks parents)."""
+def get_window_info(window):
+    """Get window's info (recursively checks parents)."""
+    wmname = window.get_wm_name()
     try:
-        wmclass = window.get_wm_class()[0]
+        wmclass = window.get_wm_class()[1]  # (process name, class name)
     except TypeError:
         wmclass = window.get_wm_class()
+        pass
     # workaround for Java app
     # https://github.com/JetBrains/jdk8u_jdk/blob/master/src/solaris/classes/sun/awt/X11/XFocusProxyWindow.java#L35
-    if not wmclass or "FocusProxy" in wmclass:
+    if not wmclass and not wmname or "FocusProxy" in wmclass:
         parent_window = window.query_tree().parent
         if parent_window:
             return get_window_info(parent_window)
-    return {'wmname': window.get_wm_name(), 'wmclass': wmclass}
+    return {'wmclass': str(wmclass), 'wmname': wmname}
+
+
+def get_display(display=None, time=30):
+    """Function to wait and get Display."""
+    while not time == 0:
+        try:
+            return Display()
+        except Xlib.error.DisplayConnectionError:
+            time =- 1
+        except Xlib.error.ConnectionClosedError as Xlib_ConnectionClosedError:
+            log_msg('Xlib: %s ' % Xlib_ConnectionClosedError)
+            return Display()
```

Comment:
What does this do? If Display just blew up on line 32, why would it suddenly be working now on line 37?  Wouldn't we want to instead fall thru to line 38 and return None?

Reply:
That's what should happen, I probably kept that line there by mistake.